viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -579,77 +579,58 @@ void SLAMServiceImpl::GetLatestSampledPointCloudMapString(
         }
     }
 
+    // Get data from painted surface in RGBA format
```

Comment:
@EshaMaharishi This is where we use the painted submaps (a cairo object) and convert it to usable pixel data for serialization.